src/IssueViz.Security/Taint/TaintList/ViewModels/TaintIssuesControlViewModel.cs

```diff
@@ -48,24 +50,55 @@ internal interface ITaintIssuesControlViewModel : IDisposable
     /// <remarks>Monitors the taint store for changes and updates the view model accordingly.</remarks>
     internal sealed class TaintIssuesControlViewModel : ITaintIssuesControlViewModel
     {
+        private readonly IActiveDocumentTracker activeDocumentTracker;
         private readonly object Lock = new object();
         private readonly INotifyCollectionChanged observableIssuesCollection;
+        private string activeDocumentFilePath;
 
         public ObservableCollection<ITaintIssueViewModel> Issues { get; } = new ObservableCollection<ITaintIssueViewModel>();
 
         public ICommand NavigateCommand { get; private set; }
 
-        public TaintIssuesControlViewModel(ITaintStore store, ILocationNavigator locationNavigator)
+        public TaintIssuesControlViewModel(ITaintStore store, 
+            ILocationNavigator locationNavigator,
+            IActiveDocumentTracker activeDocumentTracker,
+            IActiveDocumentLocator activeDocumentLocator)
         {
-
             AllowMultiThreadedAccessToIssuesCollection();
 
+            activeDocumentFilePath = activeDocumentLocator.FindActiveDocument()?.FilePath;
+            this.activeDocumentTracker = activeDocumentTracker;
+            activeDocumentTracker.OnDocumentFocused += ActiveDocumentTracker_OnDocumentFocused;
+
             var allIssues = store.GetAll();
             observableIssuesCollection = allIssues;
             observableIssuesCollection.CollectionChanged += TaintStore_CollectionChanged;
 
             UpdateIssues(allIssues, Array.Empty<IAnalysisIssueVisualization>());
             SetCommands(locationNavigator);
+            ApplyViewFilter(ActiveDocumentFilter);
+        }
+
+        private void ActiveDocumentTracker_OnDocumentFocused(object sender, DocumentFocusedEventArgs e)
+        {
+            activeDocumentFilePath = e.TextDocument?.FilePath;
+            ApplyViewFilter(ActiveDocumentFilter);
+        }
+
+        private void ApplyViewFilter(Predicate<object> filter)
+        {
+            var collectionView = CollectionViewSource.GetDefaultView(Issues);
+            collectionView.Filter = filter;
+        }
+
+        private bool ActiveDocumentFilter(object viewModel)
+        {
+            if (string.IsNullOrEmpty(activeDocumentFilePath))
+            {
+                return false;
+            }
+
+            return ((ITaintIssueViewModel)viewModel).DisplayPath.Equals(activeDocumentFilePath);
```

Comment:
File paths should be compared using `OrdinalIgnoreCase`.

Using `DisplayPath` feels like a bit hacky. It's called `DisplayPath` because there are no guarantees about what it contains, other than it's a user-friendly string. Relying on it being ,an absolute path if the relative path from the server has been resolved successfully is assuming more than the view model is promising.

At the very least this assumption should be documented.

Reply:
Good point, I changed it use `CurrentFilePath` from the underlying issue viz.